pennylane/noise/conditionals.py

```diff
@@ -0,0 +1,386 @@
+
+    Raises:
+        ValueError: if the wire set cannot be computed for ``val``.
+    """
+    iters = val if isinstance(val, (list, tuple, set, Wires)) else getattr(val, "wires", [val])
+    try:
+        wires = [[w] if isinstance(w, (int, str)) else getattr(w, "wires").tolist() for w in iters]
+    except TypeError:
+        raise ValueError(f"Wires cannot be computed for {val}") from None
+    return set(w for wire in wires for w in wire)
+
+
+def wires_in(wires):
+    """Builds a ``Conditional`` as a boolean function for evaluating
+    if a given wire exist in a specified set of wires.
+
+    Args:
+        wires (Union(list[int, str], Wires, Operation)): object to be used
+            for building the wire set.
+
+    Returns:
+        :class:`NoiseConditional <pennylane.noise.NoiseConditional>`: a boolean function
+        represented as ``WiresIn``, which evaluates to ``True`` if a given wire exist in a
+        specified set of wires.
+
+    Raises:
+        ValueError: if the wire set cannot be computed from ``wires``.
+
+    **Example**
+
+    One may use ``wires_in`` with a given sequence of wires which are used as a wire set:
+
+    >>> cond_func = qml.noise.wires_in([0, 1])
+    >>> cond_func(0)
+    True
+    >>> cond_func(3)
+    False
+
+    Additionally, if an :class:`Operation <pennylane.operation.Operation>` is provided,
+    its ``wires`` are extracted and used to build the wire set:
+
+    >>> cond_func = qml.noise.wires_in(qml.CNOT(["alice", "bob"]))
+    >>> cond_func("alice")
+    True
+    >>> cond_func("eve")
+    False
+    """
+    return NoiseConditional(
+        lambda x: _get_wires(x).issubset(_get_wires(wires)), f"WiresIn({wires})"
+    )
+
+
+def wires_eq(wires):
+    """Builds a ``Conditional`` as a boolean function for evaluating
+    if a given wire is equal to specified set of wires.
+
+    Args:
+        wires (Union(list[int, str], Wires, Operation)): object to be used
+            for building the wire set.
+
+    Returns:
+        :class:`NoiseConditional <pennylane.noise.NoiseConditional>`: a boolean function
+        represented as ``WiresEq``, which evaluates to ``True`` if a given wire is equal to
+        specified set of wires.
+
+    Raises:
+        ValueError: if the wire set cannot be computed from ``wires``.
+
+    **Example**
+
+    One may use ``wires_eq`` with a given sequence of wires which are used as a wire set:
+
+    >>> cond_func = qml.noise.wires_eq(0)
+    >>> cond_func(0)
+    True
+    >>> cond_func(qml.RY(1.23, wires=[3]))
+    False
+
+    Additionally, if an :class:`Operation <pennylane.operation.Operation>` is provided,
+    its ``wires`` are extracted and used to build the wire set:
+
+    >>> cond_func = qml.noise.wires_in(qml.RX(1.0, "dino"))
+    >>> cond_func(qml.RZ(1.23, wires="dino"))
+    True
+    >>> cond_func("eve")
+    False
+    """
+    return NoiseConditional(
+        lambda x: _get_wires(x).issubset(_get_wires(wires)), f"WiresEq({wires})"
```

Comment:
It seems to be "wires in" instead of "wires eq"

Reply:
Updated this.